realm-annotations/src/main/java/io/realm/annotations/PrimaryKey.java

```diff
@@ -25,10 +25,11 @@
  * The @PrimaryKey annotation will mark a field as a primary key inside Realm. Only one field in a
  * RealmObject class can have this annotation, and the field should uniquely identify the object.
  * Trying to insert an object with an existing primary key will result in an
- * {@link io.realm.exceptions.RealmPrimaryKeyConstraintException}.
- *
+ * {@link io.realm.exceptions.RealmPrimaryKeyConstraintException}. Primary key cannot be changed
```

Comment:
I kinda think that if you have a managed object with ID 42 and then you call `.setId(42)`, then the Realm code should detect that it is the same value and not an actual "change".